primitives/linux-port/column.cpp

```diff
@@ -1609,6 +1881,7 @@ void filterColumnData(NewColRequestHeader* in, ColResultHeader* out, uint16_t* r
   // Bit patterns in srcArray[i] representing EMPTY and NULL values
   T emptyValue = getEmptyValue<T>(dataType);
   T nullValue = getNullValue<T>(dataType);
+  uint8_t emptyValueAux = getEmptyValue<uint8_t>(execplan::AUX_COL_DATATYPE);
```

Comment:
We should use a decltype here, e.g. getEmptyValue<decltype(execplan::someAuxVarPlaceHolderVar)>(execplan::AUX_COL_DATATYPE) so that we don't go over the place if AUX Column will become uint16.

Reply:
Function calls inside `filterColumnData` such as `vectorizedFilteringDispatcher` and `scalarFiltering` will still have the `emptyValueAux` argument defined as an `uint8_t`. I am not sure how we can fix that.

Reply:
The proper way is to add another typename param for both scalarFiltering and vectorizedFilteringDispatcher to describe a AUX column integral type.